src/main/php/de/thekid/crews/web/Index.php

```diff
@@ -1,62 +1,54 @@
 <?php namespace de\thekid\crews\web;
 
-use com\mongodb\{Database, Document, ObjectId};
-use io\redis\RedisProtocol;
+use com\mongodb\{Database, Collection, Document};
 use util\Date;
-use web\frontend\{Handler, Get, Post, Delete, Put, View, Param};
+use web\frontend\{Handler, Get, Delete, Post, Param, View};
 
 #[Handler('/')]
 class Index {
+  private Collection $groups;
 
-  public function __construct(private Database $db, private RedisProtocol $pub) { }
-
-  private function post(ObjectId $id, string $view= 'post') {
-    return View::named('index')->fragment($view)->with($this->db->collection('posts')
-      ->find($id)
-      ->first()
-      ->properties()
-    );
+  public function __construct(Database $db) {
+    $this->groups= $db->collection('groups');
   }
 
   #[Get]
   public function index() {
-    $posts= $this->db->collection('posts')->aggregate([
+    $groups= $this->groups->aggregate([
       ['$sort' => ['created' => -1]],
       ['$limit' => 20],
     ]);
-    return View::named('index')->with(['posts' => $posts->all()]);
+    return View::named('index')->with(['groups' => $groups->all()]);
   }
 
-  #[Post('/posts')]
-  public function create(#[Param] string $body) {
-    $insert= $this->db->collection('posts')->insert(new Document([
-      'body'    => $body,
-      'created' => Date::now(),
-    ]));
-    $this->pub->command('PUBLISH', 'messages', "insert={$insert->id()}");
-    return View::empty()->status(204); // $this->post($insert->id());
+  #[Get('/dialog/{dialog}')]
+  public function show($dialog) {
+    return View::named('index')->fragment($dialog);
   }
 
-  #[Get('/posts/{id}/{view}')]
-  public function view(string $id, string $view) {
-    return $this->post(new ObjectId($id), $view);
+  #[Delete('/dialog/{dialog}')]
+  public function hide($dialog) {
+    return View::empty()->status(201);
```

Comment:
I dont' like this endpoint - its only purpose is to provide HTMX with an endpoint which will, when invoked, trigger removing the DOM node.

Reply:
See https://github.com/bigskysoftware/htmx/discussions/1918

Reply:
Fixed by `hx-on:click="this.closest('dialog').close()` in thekid/crews@05d744a